src/client/components/features/Alternance/Rechercher/Resultats/ListeSolutionAlternanceEntreprise.test.tsx

```diff
@@ -91,4 +91,29 @@ describe('les tags', () => {
 		expect(tags[2]).toHaveTextContent('Rencontre au sein de l’entreprise');
 		expect(tags[3]).toHaveTextContent('Candidature sur le site de l’entreprise');
 	});
+
+	describe('lien de candidature vers une entreprise', () => {
+		it("lorsque la candidature est possible, je vois le lien 'Candidater'", async () => {
+			const anEntreprisesList = [
+				aRechercheEntrepriseAlternance({
+					candidaturePossible: true,
+				}),
+			];
+
+			render(<ListeSolutionAlternanceEntreprise entrepriseList={anEntreprisesList}/>);
+
+			expect(screen.getByRole('link', { name: 'Candidater' })).toBeVisible();
+		});
+		it("lorsque la candidature n'est pas possible, je ne vois pas le lien 'Candidater'", async () => {
+			const anEntreprisesList = [
+				aRechercheEntrepriseAlternance({
+					candidaturePossible: false,
+				}),
+			];
+
+			render(<ListeSolutionAlternanceEntreprise entrepriseList={anEntreprisesList}/>);
+
+			expect(screen.queryByText('Candidater')).not.toBeInTheDocument();
+		});
```

Comment:
je propose not.toBeVisible() plutot

Reply:
J'ai essayé ` not.toBeVisible()` au départ mais le test ne passait pas 
`Error: expect(received).not.toBeVisible()

```
received value must be an HTMLElement or an SVGElement.
Received has value: null`
```

C'est parce que toBeVisible va chercher les attributes dans le DOM et vérifier si c'est display non ou opacity 0 ou hidden ou la visibilité est hidden ou collaps.

Dans notre cas, l'element est totalement pas visible, du coup le teste de visibilité fait par ToBeVisible fail car il ne trouve pas un des elements mentionné ci-dessus(invisible dans le DOM), il reçoit un 'null' car la candidature n'est pas possible pour cet offre.

C’est pourquoi j'ai mis `not.toBeInTheDocument()` ce qui vérifie la disponibilité réel d’elemnt HTML dans le DOM et pas son visibilité.

j'ai trouvé ce [thread](https://stackoverflow.com/questions/69852702/what-is-the-difference-between-tobeinthedocument-and-tobevisible) stackoverflow qui m’a aidé à comprendre  la differences entre les deux :)